docs/source/user-guide/sql/operators.md

```diff
@@ -0,0 +1,403 @@
+```
+
+### `!~`
+
+Not Regex Match
+
+```sql
+> SELECT 'datafusion' !~ '^DATAFUSION(-cli)*';
++--------------------------------------------------+
+| Utf8("datafusion") !~ Utf8("^DATAFUSION(-cli)*") |
++--------------------------------------------------+
+| true                                             |
++--------------------------------------------------+
+```
+
+### `!~*`
+
+Not Regex Case-Insensitive Match
+
+```sql
+> SELECT 'datafusion' !~* '^DATAFUSION(-cli)+';
++---------------------------------------------------+
+| Utf8("datafusion") !~* Utf8("^DATAFUSION(-cli)+") |
++---------------------------------------------------+
+| true                                              |
++---------------------------------------------------+
+```
+
+## Logical Operators
+
+- [AND](#and)
+- [OR](#or)
+
+### `AND`
+
+Logical And
+
+```sql
+> SELECT true AND true;
++---------------------------------+
+| Boolean(true) AND Boolean(true) |
++---------------------------------+
+| true                            |
++---------------------------------+
+```
+
+### `OR`
+
+Logical Or
+
+```sql
+> SELECT false OR true;
++---------------------------------+
+| Boolean(false) OR Boolean(true) |
++---------------------------------+
+| true                            |
++---------------------------------+
+```
+
+## Bitwise Operators
+
+- [& (bitwise and)](#id16)
+- [| (bitwise or)](#id17)
+- [# (bitwise xor)](#id18)
+- [>> (bitwise shift right)](#id19)
+- [<< (bitwise shift left)](#id20)
+
+### `&`
+
+Bitwise And
+
+```sql
+> SELECT 5 & 3;
++---------------------+
+| Int64(5) & Int64(3) |
++---------------------+
+| 1                   |
++---------------------+
+```
+
+### `|`
+
+Bitwise Or
+
+```sql
+> SELECT 5 | 3;
++---------------------+
+| Int64(5) | Int64(3) |
++---------------------+
+| 7                   |
++---------------------+
+```
+
+### `#`
+
+Bitwise Xor (interchangeable with `^`)
+
+```sql
+> SELECT 5 # 3;
++---------------------+
+| Int64(5) # Int64(3) |
++---------------------+
+| 6                   |
++---------------------+
+```
+
+### `>>`
+
+Bitwise Shift Right
+
+```sql
+> SELECT 5 >> 3;
++----------------------+
+| Int64(5) >> Int64(3) |
++----------------------+
+| 0                    |
++----------------------+
+```
+
+### `<<`
+
+Bitwise Shift Left
+
+```sql
+> SELECT 5 << 3;
++----------------------+
+| Int64(5) << Int64(3) |
++----------------------+
+| 40                   |
++----------------------+
+```
+
+## Other Operators
+
+- [|| (string concatenation)](#id21)
+- [@> (array contains)](#id22)
+- [<@ (array is contained by)](#id23)
+
+### `||`
+
+String Concatenation
+
+```sql
+> SELECT 'Hello, ' || 'DataFusion!';
++----------------------------------------+
+| Utf8("Hello, ") || Utf8("DataFusion!") |
++----------------------------------------+
+| Hello, DataFusion!                     |
++----------------------------------------+
+```
+
+### `@>`
+
+Array Contains
+
+```sql
+> SELECT make_array(1,2,3) @> make_array(1,3);
```

Comment:
Should we add the expected output?

```
❯ SELECT make_array(1,2,3) @> make_array(1,3);
+-------------------------------------------------------------------------+
| make_array(Int64(1),Int64(2),Int64(3)) @> make_array(Int64(1),Int64(3)) |
+-------------------------------------------------------------------------+
| true                                                                    |
+-------------------------------------------------------------------------+
```

Reply:
Thanks @alamb -- added outputs to both examples.